crates/astria-sequencer-relayer/src/data_availability.rs

```diff
@@ -355,128 +305,143 @@ impl CelestiaClient {
+                        None
+                    }
+                }
             })
             .collect::<Vec<_>>();
         Ok(sequencer_namespace_datas)
     }
 
-    /// get_sequencer_block returns the full SequencerBlock (with all rollup data) for the given
-    /// SequencerNamespaceData.
-    pub async fn get_sequencer_block(
+    /// get_sequencer_block returns the full SequencerBlock (with all rollup data) for the
+    pub async fn get_all_rollup_data_from_sequencer_namespace_data(
         &self,
-        data: &SequencerNamespaceData,
-        verification_key: Option<VerificationKey>,
+        height: u64,
+        namespace_data: &SignedNamespaceData<SequencerNamespaceData>,
     ) -> eyre::Result<SequencerBlockData> {
-        let mut rollup_txs_map = HashMap::new();
-
-        // for each rollup namespace, retrieve the corresponding rollup data
-        'namespaces: for (height, rollup_namespace) in &data.rollup_namespaces {
-            let rollup_txs = self
-                .get_rollup_data_for_block(
-                    &data.block_hash.0,
-                    *rollup_namespace,
-                    *height,
-                    verification_key,
-                )
-                .await
-                .wrap_err_with(|| {
-                    format!(
-                        "failed getting rollup data for block at height `{height}` in rollup \
-                         namespace `{rollup_namespace}`"
-                    )
-                })?;
-            let Some(rollup_txs) = rollup_txs else {
-                // this shouldn't happen; if a sequencer block claims to have written data to some
-                // rollup namespace, it should exist
-                warn!("no rollup data found for namespace {rollup_namespace}");
-                continue 'namespaces;
-            };
-            rollup_txs_map.insert(*rollup_namespace, rollup_txs);
-        }
+        let verification_key = VerificationKey::try_from(&*namespace_data.public_key)
+            .wrap_err("failed constructing verification key from stored bytes")?;
 
-        Ok(SequencerBlockData {
-            block_hash: data.block_hash.clone(),
-            header: data.header.clone(),
-            last_commit: data.last_commit.clone(),
-            rollup_txs: rollup_txs_map,
-        })
-    }
-
-    async fn get_rollup_data_for_block(
-        &self,
-        block_hash: &[u8],
-        rollup_namespace: Namespace,
-        height: u64,
-        verification_key: Option<VerificationKey>,
-    ) -> eyre::Result<Option<Vec<IndexedTransaction>>> {
+        let namespace_ids = namespace_data
+            .data
+            .rollup_namespaces
+            .iter()
+            .map(|ns| **ns)
+            .collect();
         let req = GetAllRequest {
             height,
-            namespace_ids: vec![*rollup_namespace],
+            namespace_ids,
```

Comment:
It seems like we are grabbing the data for all namespaces instead of only for a given rollup?

If so that seems to negate the purpose of storing it in different namespaces anyways? Is this for validation purposes and to be fixed @noot?

Reply:
yeah right now we grab everything, just because the merkle proofs for fetching only the relevant namespace wasn't implemented yet. we can change this to just grab the relevant namespace and do verification in a separate issue, but out of scope for this PR

Reply:
#175